src/librustc_metadata/encoder.rs

```diff
@@ -80,8 +80,8 @@ struct PerDefTables<'tcx> {
     predicates_defined_on: PerDefTable<Lazy<ty::GenericPredicates<'tcx>>>,
     super_predicates: PerDefTable<Lazy<ty::GenericPredicates<'tcx>>>,
 
-    mir: PerDefTable<Lazy<mir::Body<'tcx>>>,
-    promoted_mir: PerDefTable<Lazy<IndexVec<mir::Promoted, mir::Body<'tcx>>>>,
+    mir: PerDefTable<Lazy<mir::BodyCache<'tcx>>>,
```

Comment:
Not sure whether this represents exactly how complicated the change would be, but `rg '\bBody\b' -g'*.rs' --count-matches | awk -F ':' '{sum += $2} END {print sum}'` returns 454 when run from the root of my checkout.

Reply:
Oh but this might also be counting `hir::Body` =/. Let me be a little less lazy and see if I can figure out a better count.

Reply:
Ok, these numbers are better quick guesses:
`rg '\bBody\b' -g'*.rs' --count-matches | awk -F ':' '$1 ~ /mir/ {sum += $2} END {print sum}'`
number of times within a path containing `mir` that `Body` appears: 306
`rg "\\bmir::Body\\b|\\bBody<'" -g'*.rs' --count-matches | awk -F ':' '{sum += $2} END {print sum}'`
number of times where `mir::Body` or `Body<'` is found anywhere in the tree: 275
(`Body<'` won't match `hir::Body` because that type isn't generic over anything)

Reply:
hmm ok, in that case keeping the type makes sense I guess. The variable names should just stay at `body` though, the cached-ness is not really relevant at the use sites (like you don't name variables `rc_foo` either)